Assignments/ip1/ip1.md

```diff
@@ -0,0 +1,453 @@
+
+FakeStackOverFlow is a web application that consists of some code that runs in each client's web browser, and also code that runs on a server.
+
+This implementation effort will be split across two deliverables. In this first deliverable, you will implement and test the core backend components for this feature, and in the second deliverable, you will implement and test the frontend components. 
+
+## Objectives of this assignment
+The objectives of this assignment are to:
+*  Get you familiar with the basics of TypeScript, VSCode, and the project codebase
+*  Learn how to read and write code in TypeScript
+*  Translate high-level requirements into code
+*  Learn how to write unit tests with Jest
+
+## Getting started with this assignment
+
+Start by accepting our [invitation](https://classroom.github.com/a/ofMQDAQs). It will create a Github repository for you which will include the starter code for this assignment.
+
+
+### 1. Prerequisites
+
+#### 1.1 Install Node.js and npm
+
+If you haven't installed npm and node.js, follow the [tutorial on setting up a development environment for this class]({{site.baseurl}}{% link tutorials/week1-getting-started.md %}) 
+
+#### 1.2 Install MongoDB
+
+We use MongoDB as the NoSQL database to store data related to this application.  
+ 
+1. Follow the [instructions in the official MongoDB documentation](https://www.mongodb.com/docs/manual/administration/install-community/) to install the free community edition.
+2. Choose 'Install on Linux', 'Install on macOS', or 'Install on Windows', depending on your system.
+3. Scroll down to the section labeled 'Install MongoDB Community Edition.'and to [MongoDBDownload Center](https://www.mongodb.com/try/download/community?tck=docs_server).
+4. For Windows, in the Package dropdown, select `msi`.  Then download and run the installer.
+5. On Windows, select the _“Install MongoDB as a Service”_ checkbox and install.  This will start MongoDB as a background service.
+6. Install "MongoDB Compass" if prompted.
+7. Verify if the MongoDB server is running using the Windows Services app.
+   
+Mongo offers several methods of interacting with your Mongo databases.
+
+* MongoDB Compass is an interactive application for creating, modifying, and querying MongoDB connections.  It should be started as part of the installation process, showing a connection to `mongodb://localhost:27017/`.
+
+* Mongo shell (*mongosh*) provides a command-line interface that can be used to interact with databases in MongoDB. Download it [here](https://www.mongodb.com/try/download/shell) using the msi package. You can also use *mongosh* to see if the MongoDB server is running.  Try the MongoDB Community Edition and the command `show dbs`; you should see a list of existing databases in your local instance.
```

Comment:
Mongo shell is automatically installed with MongoDB through the Mac installation instructions. To use it, make sure MongoDB is running as a macOS service, then type `mongosh` into the terminal.

Reply:
can u update the md file if needed?